src/utils/saveFigmaDataAsNestedStructure.js

```diff
@@ -1,80 +1,99 @@
 const CONSTANT = require("../constants/constants");
 
-const saveFigmaDataAsNestedStructure = async (
-  node,
-  parent = null,
-  depth = 0,
-  currentFrameId = "",
-) => {
-  if (depth === 0 && node.type === "DOCUMENT") {
-    parent = { pages: {} };
+const saveFigmaDataAsNestedStructure = (root) => {
+  if (!root || root.type !== "DOCUMENT") {
+    return null;
   }
 
-  if (node.children) {
-    node.children.forEach((child) => {
-      saveFigmaDataAsNestedStructure(child, parent, depth + 1);
-    });
-  }
+  const queue = [{ node: root, parent: null, depth: 0, currentFrameId: "" }];
+  const structure = { pages: {} };
 
-  if (node.type === "CANVAS" && depth === CONSTANT.PAGE_NODE_DEPTH) {
-    const newPage = {
-      pageId: node.id,
-      name: node.name,
-      frames: {},
-    };
+  while (queue.length > 0) {
+    const { node, parent, depth, currentFrameId } = queue.shift();
 
-    if (parent.pages) {
-      parent.pages[newPage.pageId] = newPage;
-    }
+    let newParent = parent;
 
-    node.children.forEach((child) =>
-      saveFigmaDataAsNestedStructure(child, newPage, depth + 1),
-    );
-  } else if (node.type === "FRAME" && depth === CONSTANT.FRAME_NODE_DEPTH) {
-    const newFrame = {
-      frameId: node.id,
-      name: node.name,
-      nodes: {},
-    };
-
-    if (parent.frames) {
-      parent.frames[newFrame.frameId] = newFrame;
-    }
+    if (node.type === "CANVAS" && depth === CONSTANT.PAGE_NODE_DEPTH) {
+      const newPage = {
+        pageId: node.id,
+        name: node.name,
+        frames: {},
+      };
+
+      structure.pages[newPage.pageId] = newPage;
 
-    node.children.forEach((child) =>
-      saveFigmaDataAsNestedStructure(child, newFrame, depth + 1),
-    );
-  } else if (depth > CONSTANT.FRAME_NODE_DEPTH) {
-    const newNode = {
-      nodeId: node.id,
-      type: node.type,
-      frameId: currentFrameId,
-      property: {},
-    };
-
-    const excludedKeys = [
-      "id",
-      "name",
-      "type",
-      "scrollBehavior",
-      "blendMode",
-      "children",
-    ];
-
-    for (const key in node) {
-      if (Object.prototype.hasOwnProperty.call(node, key)) {
+      newParent = newPage;
+    } else if (node.type === "FRAME" && depth === CONSTANT.FRAME_NODE_DEPTH) {
+      const newFrame = {
+        frameId: node.id,
+        name: node.name,
+        nodes: {},
+        property: {},
+      };
+
+      const excludedKeys = [
+        "id",
+        "name",
+        "type",
+        "scrollBehavior",
+        "blendMode",
+        "children",
+      ];
```

Comment:
- `P3`
현재 `excludedKeys`를 사용하는 것을 보니
어떠한 `key`가 `excludedKeys`에 포함되어져 있는가? → `true` or `false`의 용도로만 사용되고 있는것 같습니다!

다만, true or false만 받는데 excludedKeys가 array타입이기 때문에 includes를 통해서 시간복잡도가 O(n)으로 사용이 되고 있는데요,
만약, true or false만을 판별하기 위한 모델이라면
```suggestion
      const excludedKeys = {
        id: true,
        name: true,
        type: true,
        scrollBehavior: true,
        blendMode: true,
        children: true,
      };
```

로 작성해보면 어떨까요?
이렇게 하면
기존에
```js
if (!excludedKeys.includes(key));
```
를
```js
if (!excludedKeys[key]);
```
로 대체 할수 있고, 시간복잡도 또한 `O(n)` → `O(1)`로 가져갈 수 있지 않을 까 싶습니다!

Reply:
네 해당 방식대로 객체 형태로 바꿔두었습니다. 확인 부탁드립니다 👍